P6 CL 113 Implemented small profile card component

JeffreyChuCPA opened the pull request:
This PR adds a goal buddy card component that represents the small cards seen on the main page when the user first loads, or filters for mentors, networking, etc. The main changes include:

✅ Modified component called GoalBuddyCard to render out user details in a a CSS responsive manner

✅add a function called getAllGoalBuddyData in firebase/functions to return a list of users where each user object has details from both the User Collection and Goal_Buddies Collection (No current way to retrieve all goal buddies and their respective user data)

✅add AllGoalBuddyData type to represent the object that holds the user data needed for a user from the DB

✅update ColabPage to fetch a user in the console

Test by replacing ALL code in ColabPage.tsx with the following

```
import DummyNavBar from "@/components/DummyNavBar/DummyNavBar";
import GoalBuddyCard from "@/components/GoalBuddyCard/GoalBuddyCard";
import { useEffect, useState } from "react";
import { AllGoalBuddyData } from "@/types/types";

export default function ColabPage() {
  const [goalBuddyList, setGoalBuddyList] = useState<AllGoalBuddyData[]>([])
  
  useEffect(() => {
    const fetchData = async () => {
      setGoalBuddyList([
        {
          id: 'SjAJ7Kwpf0p1BhLN28uk',
          isMentor: true,
          isNetworking: true,
          yearsOfExperience: '3-5 years',
          createdAt: {
            goalBuddy: {
              seconds: 1716922360,
              nanoseconds: 650801000,
            },
            user: {
              seconds: 1712849842,
              nanoseconds: 581744000,
            },
          },
          userId: 'SjAJ7Kwpf0p1BhLN28uk',
          googleCalendarId: 'c360884e-f250-454c-b66a-f8d3f8d04ce9',
          timezone: 'Africa/Dar_es_Salaam',
          bio: 'Sign building whatever national field garden. Lot ground than child indicate rate mean. Shoulder college music step machine truth.',
          availabilities: [],
          skills: ['Data Science', 'Web Development'],
          updatedAt: {
            goalBuddy: {
              seconds: 1724438822,
              nanoseconds: 925478000,
            },
            user: {
              seconds: 1714392710,
              nanoseconds: 923456000,
            },
          },
          isAccountabilityPartner: false,
          discipline: 'Marketing',
          country: 'Antigua and Barbuda',
          lastName: 'Cooper',
          state: 'Delaware',
          email: 'kellymiller@example.org',
          city: 'Williamview',
          username: 'wjones',
          profilePhoto: 'https://placekitten.com/484/11',
          interests: ['Data Science', 'Web Development'],
          firstName: 'Brent',
          roleId: 'B6uruTU8oncKHaun5Gwg',
        }
      ])
    }

    fetchData()
  }, [])

  return (
    <main>
      <DummyNavBar />
      <div className='flex justify-center'>
        <GoalBuddyCard goalBuddyList={goalBuddyList} />
      </div>
    </main>
  )
}
```

Redi-Molla commented:
Use Firestore's onSnapshot listener for real-time updates.
For the cards, align the name with the information below.


Everything else looks good!

KevinMeisenbacher commented:
I agree with Redi. Maybe left-aligning the names will help. Once you do that stuff, you should be good to merge!

f2xiao reviewed (commented):
this ticket is to render just individual card component, not the whole list of users

f2xiao commented:
also please include the ticket number of this pr

JeffreyChuCPA commented:
> Use Firestore's onSnapshot listener for real-time updates. For the cards, align the name with the information below.
> 
> Everything else looks good!

JeffreyChuCPA commented:
> Use Firestore's onSnapshot listener for real-time updates. For the cards, align the name with the information below.
> 
> Everything else looks good!

Not implementing onSnapshot as my function for fetching and formatting data not part of this PR. Re aligned the name and info as discussed

JeffreyChuCPA commented:
> this ticket is to render just individual card component, not the whole list of users, request to review this ticket and ticket 120 and decide if they should merge into one ticket

Removed fetch/format data function from this PR. I replaced the testing code to copy/paste to use dummy data as prop. I'd imagine ticket 120 would implement a similar fetch/format function since User collection and GoalBuddies collection are not 1 to 1 in terms of document numbers

KevinMeisenbacher commented:
In that case, what do you think of my PR where all of my code exists in my Filter component? Do you think that will be okay, or should I continue with Jeff's suggestions and move my ColabPage code from the codebase to the PR description?

JeffreyChuCPA commented:
> Tested and can confirm saw ![image](https://private-user-images.githubusercontent.com/98787046/411349253-eeca97a9-0e1c-48b1-a908-966c04da1576.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzkxMzAxNjQsIm5iZiI6MTczOTEyOTg2NCwicGF0aCI6Ii85ODc4NzA0Ni80MTEzNDkyNTMtZWVjYTk3YTktMGUxYy00OGIxLWE5MDgtOTY2YzA0ZGExNTc2LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTAyMDklMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwMjA5VDE5Mzc0NFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTE5ZjRjMDcwZjNkZTgzNDcyYmEzZGZlYTlkOTEzYWZjYWI0MTE4YzgyY2U0ZDQ4MmQwYmI4MzkzM2EzNzE4NGImWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.K-uoRYCF-7bLrScHIgC-cXvbtw-sDive0Z3fTwA820Y)
> 
> Clarification: ColabPage.tsx should have stayed unchanged? The code provided was for testing but should not be present in the final version ready for merging correct?

changed colab page to be back to what it was before.
